### ISSUE TITLE
may the tc stats test run in Python 3

### ISSUE THREAD

**Issue by burlen** [LBL-EESA/TECA#208]
```
pipin:~/TECA/bin$ctest -R py_ -I 21,21,1 --verbose
UpdateCTestConfiguration  from :/Users/bloring/TECA/bin/DartConfiguration.tcl
Parse Config file:/Users/bloring/TECA/bin/DartConfiguration.tcl
 Add coverage exclude regular expressions.
SetCTestConfiguration:CMakeCommand:/usr/local/Cellar/cmake/3.13.3/bin/cmake
UpdateCTestConfiguration  from :/Users/bloring/TECA/bin/DartConfiguration.tcl
Parse Config file:/Users/bloring/TECA/bin/DartConfiguration.tcl
Test project /Users/bloring/TECA/bin
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 69
    Start 69: py_test_tc_stats

69: Test command: /usr/local/bin/python3 "/Users/bloring/TECA/test/python/test_tc_stats.py" "/Users/bloring/TECA/../TECA_data/cam5_1_amip_run2_classify_2005_09.bin" "/Users/bloring/TECA/../TECA_data/cam5_1_amip_run2_stats_2005_09.bin"
69: Test timeout computed to be: 1500
69: ERROR: [0:0x10b1bf5c0] [/Users/bloring/TECA/python/teca_py_algorithm.h:263 2.2.1-54-g2dcb7fc]
69: ERROR: An exception ocurred when invoking the user supplied Python callback "<function teca_tc_stats.get_tc_stats_execute.<locals>.execute at 0x100ea4268>"for the execute execution phase. The exception that occurred is:
69: ERROR: [0:0x10b1bf5c0] [/Users/bloring/TECA/alg/teca_dataset_diff.cxx:90 2.2.1-54-g2dcb7fc]
69: ERROR: Input dataset 2 is NULL.
69: /usr/local/lib/python3.7/site-packages/matplotlib/font_manager.py:232: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
69:   'Matplotlib is building the font cache using fc-list. '
69: Traceback (most recent call last):
69:   File "/Users/bloring/TECA/bin/lib/teca_tc_stats.py", line 220, in execute
69:     f = open('%s_summary.csv'%(state.basename),'wc')
69: ValueError: invalid mode: 'wc'
1/1 Test #69: py_test_tc_stats .................***Failed  Error regular expression found in output. Regex=[ERROR] 28.64 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =  28.65 sec

The following tests FAILED:
	 69 - py_test_tc_stats (Failed)
Errors while running CTest
pipin:~/TECA/bin$
```

**Comment by burlen** [comment]
see also #165, #205, #206, #207

**Comment by burlen** [comment]
fixed in #203

**Comment by burlen** [comment]
see also #220
